stable/distribution/templates/distribution-statefulset.yaml

```diff
@@ -56,29 +55,29 @@ spec:
           cp -fv /tmp/etc/system.yaml {{ .Values.distribution.persistence.mountPath }}/etc/system.yaml;
           echo "Remove {{ .Values.distribution.persistence.mountPath }}/lost+found folder if exists";
           rm -rfv {{ .Values.distribution.persistence.mountPath }}/lost+found;
-        {{- if or .Values.distribution.joinKey .Values.distribution.joinKeySecretName }}
+        {{- if or .Values.distribution.joinKey .Values.distribution.joinKeySecretName .Values.global.joinKey .Values.global.joinKeySecretName }}
           echo "Copy joinKey to {{ .Values.distribution.persistence.mountPath }}/etc/security";
           mkdir -p {{ .Values.distribution.persistence.mountPath }}/etc/security;
           echo ${DISTRIBUTION_JOIN_KEY} > {{ .Values.distribution.persistence.mountPath }}/etc/security/join.key;
         {{- end }}
-        {{- if or .Values.distribution.masterKey .Values.distribution.masterKeySecretName }}
+        {{- if or .Values.distribution.masterKey .Values.distribution.masterKeySecretName .Values.global.masterKey .Values.global.masterKeySecretName }}
           echo "Copy masterKey to {{ .Values.distribution.persistence.mountPath }}/etc/security";
           mkdir -p {{ .Values.distribution.persistence.mountPath }}/etc/security;
           echo ${DISTRIBUTION_MASTER_KEY} > {{ .Values.distribution.persistence.mountPath }}/etc/security/master.key;
         {{- end }}
         env:
-        {{- if or .Values.distribution.joinKey .Values.distribution.joinKeySecretName }}
+        {{- if or .Values.distribution.joinKey .Values.distribution.joinKeySecretName .Values.global.joinKey .Values.global.joinKeySecretName }}
         - name: DISTRIBUTION_JOIN_KEY
           valueFrom:
             secretKeyRef:
-              name: "{{ .Values.distribution.joinKeySecretName | default (include "distribution.fullname" .) }}"
+              name: {{ include "distribution.joinKeySecretName" . }}
               key: join-key
         {{- end }}
         {{- if or .Values.distribution.masterKey .Values.distribution.masterKeySecretName }}
```

Comment:
You didn’t add the global values to this if

Reply:
Fixed